mesosphere/kudo-cassandra-operator
# Custom Cassandra YAML and JVM options.

## What changes are proposed in this PR? 
Resolves  [DCOS-59496](https://jira.mesosphere.com/browse/DCOS-59496)
Resolves  [DCOS-59495](https://jira.mesosphere.com/browse/DCOS-59495) 

This PR Implement adding:

- Base64 encoded Cassandra properties appended to cassandra.yaml.
- Base64 encoded JVM Options appended to jvm.options.

Implementation steps : 
1. Added `CUSTOM_CASSANDRA_YAML_BASE64` and `CUSTOM_JVM_OPTIONS_BASE64` properties to `params.yml`
2. Modified `jvm-options.yaml` and `cassandra-yaml.yaml`. 
3. Used [b64dec](http://masterminds.github.io/sprig/) template function to decode base64 property.
## How were these changes tested?
Manually tested  
1. Install operator 
2. Look into pod `cassandra-node-0` from k8s ui. On `exec` console `cat` the content of `etc/cassadra/cassanda.yaml` and `etc/cassadra/jvm.options` and see that properties were decoded and propagated.
